docs/proposals/27-falco-extension.md

```diff
@@ -0,0 +1,352 @@
+        address: <http...>
+        customHeaders: ...
+        # check if ssl certificate of the output is valid
+        checkCerts: ...
+```
+
+In a simple setup the following default configuration is sufficient to
+deploy Falco:
+
+```
+  - providerConfig:
+      apiVersion: falco.extensions.gardener.cloud/v1alpha1
+      kind: FalcoServiceConfig
+    type: shoot-falco-service
+```
+
+In this case the extension will deploy the current Falco version and will 
+take care of automated updates. The ruleset will be managed by Gardener and 
+will only provide the standard rules which are designed to offer basic 
+detection capabilities but to avoid too many false positives. Visualization 
+of events will be provided in the Gardener dashboard.
+
+This will correspond to the following configuration which will be reflected
+in the shoot manifest:
+
+```
+  - providerConfig:
+      apiVersion: falco.extensions.gardener.cloud/v1alpha1
+      kind: FalcoServiceConfig
+      falcoVersion: <latest Falco version marked as "supported">
+      falcoAutoUpdate: true
+      resources: gardener
+      gardener:
+        useFalcoRules: true
+        useFalcoIncubatingRules: false
+        useFalcoSandboxRules: false
+    type: shoot-falco-service
+```
+
+Falco versions are maintained by the extension in the same way operating
+system-, and Kubernetes versions are maintained by Gardener. If no 
+version is specified the Falco extension will select the currently supported
+version.
+
+There are two methods for managing rules and plugins: either use setup 
+provided by Gardener by setting `resources` to `gardener` or use 
+`falcoctl` by setting `resources` to `falcoctl`.
+
+Custom rules in the Gardener setup can be stored in a ConfigMap and applied 
+to the same project as the cluster. The ConfigMap must be configured as a 
+resource in the cluster as described in [6]:
+
+```
+  resources:
+  - name: customRules
+    resourceRef:
+      apiVersion: v1
+      kind: ConfigMap
+      name: my-custom-rules
+```
+
+This is a template ConfigMap which contains the custom rules. They will be
+deployed and ordered based on the result of the Go programming language
+`strings.Compare` function.
+
+```
+kind: ConfigMap
+  name: my-custom-rules
+  namespace: garden-myproject
+data:
+  myrules.yaml: |-
+    # custom rules go here
+    ...
+```
+
+Events that are generated by Falco are forwarded to Falcosidekick that is 
+deployed alongside Falco. The Falcosidekick webhook output is configured
+to send events to a central location. In the default configuration it will be
+the [Falco event ingestor](#falco-event-ingestor).
+Falcosidekick webhooks are not to be confused with Kubernetes webhooks, they
+are used to post events to arbitrary locations identified by a URL, see 
+[Webhook](https://github.com/falcosecurity/falcosidekick/blob/master/docs/outputs/webhook.md).
+for its configuration. The Falco extension issues JWT tokens containing the
+cluster identity.
```

Comment:
What are the features of the JSON web token? How are the receivers verifying them?